src/TopRefNodeParser.ts

```diff
@@ -12,16 +12,22 @@ export class TopRefNodeParser implements NodeParser {
 
     public createType(node: ts.Node, context: Context): BaseType | undefined {
         const baseType = this.childNodeParser.createType(node, context);
+        const sourceFileName = node.getSourceFile().fileName;
 
         if (baseType === undefined) {
             return undefined;
         }
 
         if (this.topRef && !(baseType instanceof DefinitionType)) {
-            return new DefinitionType(this.fullName, baseType);
+            const defBaseType = new DefinitionType(this.fullName, baseType);
+            defBaseType.sourceFileName = sourceFileName;
```

Comment:
This doesn't feel clean. Can we make `sourceFileName` an argument to the `DefinitionType` constructor?

Reply:
Thank you for your correction. I think it is a very good suggestion. I will try to submit a version again tomorrow to correct these problems.

Reply:
Thank you for your pull request.

Reply:
Any updates @charlzyx?

Reply:
> 

Sorry i'm too late, too busy to submit; And sorry again, i don't know how to submit this or, so i have reopen another one #1056

Reply:
Can you push to the branch this pull request was made from?

Reply:
sorry about #1056  this is my first time to use github pr，I'm not familiar with it.
just want submit a cleaning commit，i will try fix next free time.

Reply:
Okay. Please notify me when you want a review then.

Reply:
i'm sorry but i have to say a bad info:  this PR's fork in my personal repo is deleted, and not local repo. so link [charlzyx:next] override by #1056 's fork. and this PR cannot edit.

![image](https://user-images.githubusercontent.com/18055018/145130502-513499ba-01cf-4b49-b6fe-7b8444ad848f.png)

![image](https://user-images.githubusercontent.com/18055018/145130472-037a9e21-b653-4591-a288-0e7a0d8b1228.png)

If possible, I suggest closing the current PR and reopening #1056

Reply:
Okay, please make sure my comment here are addressed. Ideally, copy them over.